scripts/perfmon

```diff
@@ -693,6 +705,49 @@ class VMMonitor(ObjectMonitor):
         elif config_tag == 'alarm_priority':             return '3'   # Service degradation level defined in PR-1455
         else: raise XmlConfigException, "variable %s: no default available for tag %s" % (variable_name, config_tag)
 
+class SRMonitor(ObjectMonitor):
+    """Object that maintains state of one SR
+
+    Configured by writing an xml string into an other-config key, e.g.
+    xe sr-param-set uuid=$vmuuid other-config:perfmon=\
+       '<config><variable><name value="physical_utilisation"/><alarm_trigger_level value="0.8"/></variable></config>'
+
+    Notes:
+     - Multiple <variable> nodes allowed
+     - full list of child nodes is
+       * name: what to call the variable (no default)
+       * alarm_priority: the priority of the messages generated (default '3')
+       * alarm_trigger_level: level of value that triggers an alarm (no default)
+       * alarm_trigger_sense: 'high' if alarm_trigger_level is a max, otherwise 'low'. (default 'high')
+       * alarm_trigger_period: num seconds of 'bad' values before an alarm is sent (default '60')
+       * alarm_auto_inhibit_period: num seconds this alarm disabled after an alarm is sent (default '3600')
+       * consolidation_fn: how to combine variables from rrd_updates into one value
+         (default is 'get_percent_sr_usage' for 'physical_utilistation', & 'sum' for everything else)
+       * rrd_regex matches the names of variables from (xe sr-data-sources-list uuid=$sruuid) used to compute value
+         (has default for "physical_utilistaion")
+    """
+    def __init__(self, *args):
+        self.monitortype = "SR"
+        ObjectMonitor.__init__(self, *args)
+        print_debug("Created SRMonitor with uuid %s" % self.uuid)
+
+    def get_default_variable_config(self, variable_name, config_tag):
+        "This allows user to not specify full set of tags for each variable in xml config"
+        if config_tag == 'consolidation_fn':
+            if variable_name == "physical_utilisation": return 'get_percent_sr_usage'
```

Comment:
why is "physical_utilisation" double quoted when all other strings here are single quoted?